OrbitVulkanLayer/CommandBufferManager.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitVulkanLayer/CommandBufferManager.h"
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit::layer {
+
+void CommandBufferManager::TrackCommandPool(VkCommandPool pool) {
+  absl::WriterMutexLock lock(&mutex_);
+  tracked_pools_.insert(pool);
+}
+
+void CommandBufferManager::UnTrackCommandPool(VkCommandPool pool) {
+  absl::WriterMutexLock lock(&mutex_);
+  CHECK(IsCommandPoolTracked(pool));
+  tracked_pools_.erase(pool);
```

Comment:
It's all behind a lock here, so it doesn't really matter, but it seems the more natural order would be to do this last (when all associated data has been removed), or is there a reason you need this order?